sktime/base/_poly.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements polymorphic base class."""
+
+__author__ = ["fkiraly", "miraep8"]
+__all__ = ["BasePolymorph"]
+
+
+from sktime.base import BaseObject
+from sktime.registry._lookup import _check_estimator_types
+
+
+class BasePolymorph(BaseObject):
+    """Handles parameter management for estimators composed of named estimators.
+
+    Partly adapted from sklearn utils.metaestimator.py.
+    """
+
+    def __init__(self, estimator_type="base"):
+
+        self.estimator_type = estimator_type
+        baseclass = _check_estimator_types(estimator_type)[0]
+
+        super(BasePolymorph, self).__init__()
+
+        attr_to_copy = set(dir(baseclass)).difference(dir(self))
+
+        for attr in attr_to_copy:
+            setattr(self, attr, getattr(baseclass, attr))
+
+        # baseclass.__init__(self)
+
+
+def _delegate(func):
```

Comment:
I am thinking for this to work you would also need to pass/take self?  Something that might be worth trying actually could simply be getting the function from the delegated class using `get_attr` (functions are also attributes after all!) This could be cleaner/easier to do via decorators?  (ie functions that should delegated should be decorated!).  But this is just one way to achieve this.

Reply:
I was thinking repeatedly that `_delegate` may need to take `self`, but *does it*? `func` has `self` as an argument, so all is fine, I thought. Then, it suffices that `inner` has `self` as arg.

I don't entirely see how it would look like with `get_attr`?

Further, the issue that remains unsolved is properties.

Reply:
Sure, if it works it works.

Reply:
I don't know whether it works because I'm stuck on the properties

Reply:
the solution I arrived at now does not use delegation